src/app/code/community/Shopgate/Cloudapi/Helper/Frontend/Checkout.php

```diff
@@ -78,20 +76,26 @@ public function logoutCustomer()
     }
 
     /**
-     * @param int        $resourceId
-     * @param int | null $customerId
+     * @param int          $resourceId
+     * @param int | string $storeId
```

Comment:
generateToken is only called in Shopgate_Cloudapi_Model_Api2_Carts_Url_Rest::_createUrl, whereas the $storeId will be returned by Mage_Api2_Model_Resource::_getStore. This method is returning an instance of Mage_Core_Model_Store. Could you please double check?